tensorflow/contrib/framework/python/ops/variables_test.py

```diff
@@ -810,5 +810,27 @@ def testInitFromCheckpointWithScopes(self):
       self.assertAllEqual(init_value0, var0.eval())
       self.assertAllEqual(init_value1, var1.eval())
 
+class ZeroInitializerOpTest(tf.test.TestCase):
+
+  def _testZeroInitializer(self, shape, dtype):
+    var0 = tf.Variable(tf.zeros(shape, dtype=dtype))
+    var1 = tf.Variable(tf.ones(shape, dtype=dtype))
+    var0_zero = tf.contrib.framework.zero_initializer(var0)
+    var1_zero = tf.contrib.framework.zero_initializer(var1)
```

Comment:
It doesn't have to if zero_initializer is the op you wrote; the op passed to tf.Variable is just added to the initializers graph collection so it is run by tf.initialize_all_variables(). If it doesn't create any additional storage (and the op you wrote doesn't) then you should be fine. 

I also think we should test the memory usage by turning tracing on in the RunOptions and examining the step_stats.dev_stats.node_stats.allocator_used_memory.peak_bytes. That way the unit test will protect the code from changes which break this important property.